typewriter/fixes/tests/test_util.py

```diff
@@ -0,0 +1,114 @@
+                         (None, "a", "x", "import b as x"),
+                         (None, "a", "x", "import b as x, c, d as y"),
+                         (None, "y", "b", "from x import y as a"))
+        for package, name, alias, import_ in failing_tests:
+            n = self.get_import_binding_info(import_, name, package)
+            if n:
+                # If valid binding, check to make sure import info is correct
+                self.assertFalse(n.binding_name == alias and n.import_name == name)
+            else:
+                # Should return None
+                self.assertFalse(n)
+
+        passing_tests = ((None, "a", "z", "from x import a as z"),
+                         (None, "a", "z", "import a as z"),
+                         (None, "a.b", "z", "import a.b as z"),
+                         ("a", "b", "z", "import a.b as z"),
+                         ("a.b", "c", "z", "import a.b.c as z"),
+                         (None, "a", "z", "import a as z\nimport u as v"),
+                         (None, "a", "w", "from x import b as y, c as z, a as w, d"),
+                         (None, "a", "z", "from x.b import a as z"),
+                         (None, "a", "z", "from x.b import b as y, c as w, a as z, d"),
+                         (None, "a", "z", "import b as x, c as y, a as z, d as w"))
+        for package, name, alias, import_ in passing_tests:
+            n = self.get_import_binding_info(import_, name, package)
+            self.assertTrue(n)
+            self.assertTrue(n.import_name == name)
+            self.assertTrue(n.binding_name == alias)
+
+    def test_in_function(self):
+        self.try_with("def foo():\n\tbar.baz()\n\tstart=3")
+
+
+class Test_find_import_info(TestCase):
+
+    def find_import_info(self, name, string, package):
+        node = parse(string)
+        print(node)
+        # Return the node as an import binding if possible, None if otherwise
+        return fixer_utils.find_import_info(name, node, package)
+
+    def try_with(self, string):
+        failing_tests = (("a", "a", "x", "from a import b as x"),
+                         ("a.d", "a", "x", "from a.d import b as x"),
+                         ("d.a", "a", "x", "from d.a import b as x"),
+                         (None, "a", "x", "import b as x"),
+                         (None, "a", "x", "import b as x, c, d as y"),
+                         (None, "y", "b", "from x import y as a"))
+        for package, name, alias, import_ in failing_tests:
+            n = self.find_import_info(name, import_ + '\n' + string, package)
+            if n:
+                # If valid binding, check to make sure import info is correct
+                self.assertFalse(n.binding_name == alias and n.import_name == name)
+            else:
+                # Should return None
+                self.assertFalse(n)
+
+        passing_tests = ((None, "a", "z", "from x import a as z"),
+                         (None, "a", "z", "import a as z"),
```

Comment:
I'd still like to see the "simple" base case tests, ie:

```
import a
import a.b
import a.b.c
```

https://github.com/chadrik/typewriter/pull/2#discussion_r547465815